### DISCUSSION TITLE
About the function of the ARC module

### DISCUSSION THREAD

**Discussion by Eli-question** [rspamd/rspamd#3946] [Q&A]
Hey, 

[ARC module](https://rspamd.com/doc/modules/arc.html) has the following description.
```
# Domain to use for ARC signing: can be "header" or "envelope"
use_domain = "header";
```

Please tell me the difference between "header" and "envelope".

**Comment by kasak87** [comment]
Have you ever send a message throught telnet session? 
When you connect to server, first you say hello to it, after you must enter 
**mail from:** and **rcpt to:** directives, which mail server use to allow or deny relaying. 
After this, you start typing the body of message, that can itself contain **to:** and **from:** headers. 
The first step when you enter **rcpt to:** and mail **from:** is the envelope of message. 
and if the **from:** and **to:** headers is written to body of email, this is "header"
similarly, if you send real paper message through your local mail office, you can write any information on the envelope
and inside your mail, you can write any infomation on your paper mail itself.

**Reply by Eli-question** [reply]
Thanks for your answer!

I would like to get a better understanding of ARC signing and Email spoofing.
My understanding is that spoofed email is written disinformation in "header", so I think it's better to sign "envelope" used sending and receiving emails to server for ARC signing. 

Could you tell me benefits of using "header"?

**Reply by kasak87** [reply]
honestly, i'm not very experienced in large relay networks, maybe header is somewhere usable. I'm using arc signing with envelope, and never had any problems with it.

**Reply by Eli-question** [reply]
I’m sorry for the late reply.

I see. Thank you for your response;)
I'll comment on this page when I find anything on this.

Thanks!!
